lua/weapons/weapon_spraypaint/cl_init.lua

```diff
@@ -1,66 +1,576 @@
+    local normal = net.ReadVector()
+    local color = net.ReadColor()
+    local size = net.ReadFloat()
+    local ent = net.ReadEntity()
+    util.DecalEx(GetPaintMaterial(color), ent, pos, normal, color, size, size)
+end)
+
+local wepicon = Material("spraypaint/spraypaint_icon.png", "smooth")
+
+function SWEP:DrawWeaponSelection(x, y, wide, tall, alpha)
+    self.SelNeeded = 5
+
+    if (not IsValid(self.TempModel)) then
+        self.TempModel = ClientsideModel(self.WorldModel)
+        self.TempModel:SetNoDraw(true)
+    else
+        local pan = Angle(0, 0, 0)
+        pan:RotateAroundAxis(Vector(0, 1, 0), -25)
+        pan:RotateAroundAxis(Vector(0, 0, 1), CurTime() * 270)
+        self.TempModel:SetAngles(pan)
+    end
+
+    if (IsValid(self.TempModel)) then
+        cam.Start2D()
+        cam.Start3D(Vector(0, -37, 1), Angle(0, 90, 0), 32, x, y, wide, tall)
+        render.SuppressEngineLighting(true)
+
+        for i = 0, 5 do
+            render.SetModelLighting(i, 0.5, 0.5, 0.5)
+        end
+
+        render.SetModelLighting(4, 1, 1, 1)
+        render.SetModelLighting(5, 0, 0, 0)
+        local cl = self:GetCustomColor() or Vector(1, 0, 1)
+        render.SetColorModulation(cl.x, cl.y, cl.z)
+        self.TempModel:DrawModel()
+        render.SuppressEngineLighting(false)
+        cam.End3D()
+        cam.End2D()
+        local shit = self.TempModel
+        timer.Remove("WEPSEL_SPRAYPAINT_" .. self.TempModel:EntIndex())
+
+        timer.Create("WEPSEL_SPRAYPAINT_" .. self.TempModel:EntIndex(), 0.2, 1, function()
+            if (IsValid(shit)) then
+                shit:Remove()
+            end
+        end)
+    end
+end
+
+function SWEP:PreDrawViewModel(vm, weapon, ply)
+    self:SetBodygroup(2, (self.CapOn and 0) or 1)
+    local cl = self:GetCustomColor()
+    render.SetColorModulation(cl.x, cl.y, cl.z)
+    local down = ply:KeyDown(IN_ATTACK)
+    self.FingerLerp = self.FingerLerp or 0
+
+    if (down) then
+        self.FingerLerp = math.min(self.FingerLerp + FrameTime() * 12, 1)
+    end
+
+    if (not down) then
+        self.FingerLerp = math.max(self.FingerLerp - FrameTime() * 8, 0)
+    end
+
+    vm:SetPoseParameter("press", self.FingerLerp)
+end
+
+hook.Add("PostDrawViewModel", "paint_handsfix", function(vm, ply, weapon)
+    render.SetColorModulation(1, 1, 1)
+end)
+
+function SWEP:PostDrawViewModel(vm, weapon, ply)
+    render.SetColorModulation(1, 1, 1)
+end
+
+function SWEP:DrawWorldModel(query)
     local ply = self:GetOwner()
+    self:SetModelScale(1, 0)
+    self:SetSubMaterial()
+    local matrix
+    local horn = false
+    local opos = self:GetPos()
+    local oang = self:GetAngles()
 
-    if (IsValid(ply)) then
-        local bn = ply:IsPony() and "LrigScull" or "ValveBiped.Bip01_R_Hand"
+    if IsValid(ply) then
+        local modelStr = ply:GetModel():sub(1, 17)
+        local isPony = modelStr == "models/ppm/player" or modelStr == "models/mlp/player" or modelStr == "models/cppm/playe"
```

Comment:
why copy the code that already exists in player_extension.lua, just call ply:IsPony

Reply:
i think this is leftover from old code.